Kernel/Term.cpp

```diff
@@ -678,14 +679,76 @@ std::string Term::toString(bool topLevel) const
     printArgs = isSort() || env.signature->getFunction(_functor)->combinator() == Signature::NOT_COMB;
   }
 
-  std::string s = headToString();
+#if NICE_THEORY_OUTPUT
+  auto theoryTerm = Kernel::tryNumTraits([&](auto numTraits) {
```

Comment:
Would it make sense to move this to `NumTraits`?

Reply:
Hmmm i don't really think it would fit there. But I could factor it out and move it to a separate file if that's nicer you think.